p2p_wallet/Scenes/Main/Swap/ChooseSwapItem/ChooseSwapTokenCoordinator.swift

```diff
@@ -30,11 +30,11 @@ final class ChooseSwapTokenCoordinator: Coordinator<SwapToken?> {
             service: ChooseSwapTokenService(swapTokens: tokens, fromToken: fromToken),
             chosenToken: chosenWallet
         )
-        let view = ChooseItemView<ChooseSwapTokenItemView>(viewModel: viewModel) { [unowned self] model in
+        let view = ChooseItemView<ChooseSwapTokenItemView>(viewModel: viewModel) { [weak self] model in
             ChooseSwapTokenItemView(
                 token: model.item as! SwapToken,
                 chosen: model.isChosen,
-                fromToken: fromToken
+                fromToken: self?.fromToken ?? true
```

Comment:
if we need only the fromToken, we can just copied it outside the closure and get rid of capturing self
```swift
let fromToken = fromToken
let view = ...
```

Reply:
oh ok, nice idea

Reply:
fixed

Reply:
Approved. Thanks! 👍